core/session.ts

```diff
@@ -284,7 +284,7 @@ export class Session {
     return newTable;
   }
 
-  public assert(name: string, query?: AContextable<string>): Assertion {
+  public assert(name: string, query?: AContextable<string>, parent?: string): Assertion {
```

Comment:
IMO this feels like a bad idea. I would suggest setting the `parent` field of the `Assertion`'s protobuf directly inside the `config()` method in `table.ts`. something like `assert(...).proto.parent = this.proto.name`